src/Handler/DeeplBatchTranslationRequestHandler.php

```diff
@@ -44,37 +40,33 @@ public function getPath(): string
 
     public function getBody(): StreamInterface
     {
-        $query = http_build_query(
-            array_filter(
-                [
-                    'target_lang' => $this->translation->getTargetLang(),
-                    'tag_handling' => implode(
-                        self::SEPARATOR,
-                        $this->translation->getTagHandling()
-                    ),
-                    'non_splitting_tags' => implode(
-                        self::SEPARATOR,
-                        $this->translation->getNonSplittingTags()
-                    ),
-                    'ignore_tags' => implode(self::SEPARATOR, $this->translation->getIgnoreTags()),
-                    'split_sentences' => $this->translation->getSplitSentences(),
-                    'preserve_formatting' => $this->translation->getPreserveFormatting(),
-                    'glossary_id' => $this->translation->getGlossaryId(),
-                    'auth_key' => $this->authKey,
-                ]
-            )
-        );
+        $body = array_filter([
+            'target_lang' => $this->translation->getTargetLang(),
+            'tag_handling' => implode(
+                self::SEPARATOR,
+                $this->translation->getTagHandling(),
+            ),
+            'non_splitting_tags' => implode(
+                self::SEPARATOR,
+                $this->translation->getNonSplittingTags(),
+            ),
+            'ignore_tags' => implode(self::SEPARATOR, $this->translation->getIgnoreTags()),
+            'split_sentences' => $this->translation->getSplitSentences(),
+            'preserve_formatting' => $this->translation->getPreserveFormatting(),
+            'glossary_id' => $this->translation->getGlossaryId(),
+            'text' => [],
+        ]);
 
         // add the text parameters separately as http_build_query would create `text[]` params
         foreach ($this->translation->getText() as $text) {
-            $query .= '&text=' . $text;
+            $body['text'][] = $text;
         }
 
-        return $this->streamFactory->createStream($query);
+        return $this->streamFactory->createStream(json_encode($body, JSON_THROW_ON_ERROR));
     }
 
     public function getContentType(): string
     {
-        return 'application/x-www-form-urlencoded';
+        return 'application/json';
```

Comment:
if this is now application/json everywhere - this can also be removed in the handler or/and the overwriting in the individual handlers see FileRequestHandler/others

Reply:
> if this is now application/json everywhere - this can also be removed in the handler or/and the overwriting in the individual handlers see FileRequestHandler/others

:+1: I'd rather move it into AbstractDeeplHandler if everyhting is json now. Single Handlers can overwrite it if necessary